### PR TITLE
Added Contract Automation

### PR DESCRIPTION
**Pull request by IllianiBird**
Currently accepting a contract is a bit of a process: you need to accept the contract, navigate to the Briefing Room, click on the system, click on Calculate Jump Path, click on Begin Transit. Then, by the time you arrive half of your forces are falling apart from failed maintenance checks.

To resolve that last problem, I'm aware that many of our players advise mothballing forces before transit. Seeing just how crippling RAW maintenance rules can be, I'm inclined to agree.

From a New Player Experience perspective, both of these issues present a couple of problems:
- Having beginning a contract such a process creates multiple 'quit moments' for the player, multiple stumbling blocks where if they can't figure out what they want to do they're potentially just going to close the program.
- Players don't expect to arrive with half of their 'Meks falling apart. From what I've seen, this usually results in the player just GM fixing the units, or turning off maintenance entirely.
- Mothballing units means the player has to remember to unmothball them when they arrive. If they forget, they're likely to just GM Active.

This PR attempts to resolve these options in a couple of steps:
- When the contract is accepted the player is presented with a dialog asking whether they would like to mothball their TO&E for transport. This mothballing is automatic, to avoid any issues with Tech assignment. We're essentially GM Mothballing, and this can be justified as the employer hiring techs or as techs to ensure their contracted unit is able to get en route as quickly as possible.
- A second dialog is then triggered, asking the player whether they would like to begin transit to the contract system. If they accept a route is immediately laid out and transit begins.
- When the contract date arrives all of the players units, previously mothballed by this system, are automatically un-mothballed.

### CODE REVIEW

**Comment by WeaverThree** [comment]
I've always just run with x4 maint time and it hasn't been a problem during transit, buuuut I'm also using 'only break on A quality' and I don't know if I'm running with the default maintenence values.

One thing I can't tell from glancing at the code, is does this mothball units currently under refit? 

Right now, I just checked, you can start a refit and then mothball a unit, and the refit will proceed. But you cannot refit a unit that is already mothballed. Does this need to change?

**Comment by IllianiCBT** [comment]
My gut tells me we probably shouldn't allow refits on mothball units at all, and we should probably prevent the mothballing of units undergoing refit.

**Comment by IllianiCBT** [comment]
Updated to check unit is both available (is present, not deployed, not refitting, not mothballing, and not mothballed) and not under repairs.

**Review by rjhancock — APPROVED** [review]
Just did a quick code review.

**Comment by IllianiCBT** [comment]
I appreciate you.

**Review by SJuliez — COMMENTED** [review]
Great idea this!

File: MekHQ/src/mekhq/campaign/market/contractMarket/ContractAutomation.java [hunk 1]
```diff
@@ -0,0 +1,338 @@
...
+        description.setBorder(BorderFactory.createTitledBorder(descriptionTitle));
+        descriptionPanel.add(description);
+
+        // Create main JPanel and add icon and description
+        JPanel mainPanel = new JPanel(new BorderLayout());
+        mainPanel.add(iconLabel, BorderLayout.NORTH);
+        mainPanel.add(descriptionPanel, BorderLayout.CENTER);
+
+        // Create JOptionPane
+        JOptionPane optionPane = new JOptionPane(mainPanel,
```
[the remark this answers is not in the document]
Comment (IllianiCBT) [reply MekHQ/src/mekhq/campaign/market/contractMarket/ContractAutomation.java:249]: That was the intent, but on a second viewing I don't think it's necessary and double-dialoging isn't doing anything beneficial here, just unnecessarily increasing complexity.

Removed.

File: MekHQ/src/mekhq/campaign/market/contractMarket/ContractAutomation.java [hunk 2]
```diff
@@ -0,0 +1,338 @@
...
+
+        // This needs to be a separate list as the act of mothballing the unit removes it from the
+        // list of units attached to the relevant force, resulting in a ConcurrentModificationException
+        for (Unit unit : mothballTargets) {
+            mothballUnitAction.execute(campaign, unit);
+            MekHQ.triggerEvent(new UnitChangedEvent(unit));
+            mothballedUnits.add(unit);
+        }
+
+        return mothballedUnits;
```
[the remark this answers is not in the document]
Comment (IllianiCBT) [reply MekHQ/src/mekhq/campaign/market/contractMarket/ContractAutomation.java:311]: Good point well made

**Comment by IllianiCBT** [comment]
Updated to incorporate suggestions from Julie

**Pull request merged**